nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java

```diff
@@ -290,32 +285,34 @@ public void renameTable(TableIdentifier from, TableIdentifier toOriginal) {
   @Override
   public void createNamespace(Namespace namespace, Map<String, String> metadata) {
     try {
-      reference.checkMutable();
-      api.createNamespace()
-          .reference(reference.getReference())
+      client.getRef().checkMutable();
+      client.getApi().createNamespace()
```

Comment:
It seems to me that the client should expose `createNamespace()` that does the check above and exposes this API? It seems weird that this is basically asking the client for its internal state and modifying that state (violates "tell don't ask").

Reply:
I agree. my initial approach was to just move a few things over to the `NessieIcebergClient`, but now it makes perfect sense to have most of the code in the client itself, which is then just used by the catalog